### PR TITLE
Update details on JAX libraries in JAX README.md

### PR DESCRIPTION
**Pull request by 8bitmp3**
@jakevdp

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -411,23 +411,17 @@ community-supported conda build, and answers to some frequently-asked questions.
...
-foundation for several other libraries in the JAX ecosystem.
-
-In addition, DeepMind has open-sourced an [ecosystem of libraries around
-JAX](https://deepmind.com/blog/article/using-jax-to-accelerate-our-research)
-including [Optax](https://github.com/deepmind/optax) for gradient processing and
-optimization, [RLax](https://github.com/deepmind/rlax) for RL algorithms, and
-[chex](https://github.com/deepmind/chex) for reliable code and testing. (Watch
+[Flax](https://github.com/google/flax) and its [documentation site](https://flax.readthedocs.io/en/latest/nnx/index.html).
+
+Check out the [JAX Ecosystem section](https://jax.readthedocs.io/en/latest/#ecosystem)
```
Comment (jakevdp) [inline README.md:419]: I think we should still call out equinox here

Comment (8bitmp3) [reply README.md:419]: OK :+1:

Comment (8bitmp3) [reply README.md:419]: Done. LMKWYT.

Comment (8bitmp3) [reply README.md:419]: Line 423.

**Review by jakevdp — APPROVED** [review]
Thanks!

**Pull request merged**